src/server/internalServer.cpp

```diff
@@ -714,6 +714,12 @@ std::unique_ptr<Response> InternalServer::handle_search(const RequestContext& re
 
   try {
     if (request.get_url_part(1) == "searchdescription.xml") {
+      // Be sure there is not url_part 2
+      try {
+        request.get_url_part(2);
+        return HTTP404Response(*this, request)
+         + urlNotFoundMsg;
+      } catch (std::out_of_range&) {}
       auto response = ContentResponse::build(*this, RESOURCE::ft_opensearchdescription_xml, get_default_data(), "application/opensearchdescription+xml");
```

Comment:
Won't it be simpler to handle "/search/searchdescription.xml" as a special case in `handle_request()`?

Reply:
Indeed